File: ros2_ws/src/hello_ros2_pkg/hello_ros2_pkg/talker.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String # 기본 문자열 메시지 타입

class Talker(Node):
    def __init__(self):
        super().__init__('talker') # 노드 이름을 'talker'로 초기화
        self.publisher_ = self.create_publisher(String, 'hello_topic', 10)
        self.timer = self.create_timer(0.5, self.timer_callback) # 0.5초마다 실행
        self.i = 0

    def timer_callback(self):
        msg = String()
        msg.data = f'Hello ROS2! [{self.i}]'
        self.publisher_.publish(msg) # 메시지 발행
        self.get_logger().info(f'Publishing: "{msg.data}"') # 로그 출력
        self.i += 1

def main(args=None):
    rclpy.init(args=args) # ROS2 초기화
    node = Talker()
    rclpy.spin(node) # 노드 실행 유지
    node.destroy_node() # 종료 시 노드 파괴
    rclpy.shutdown() # ROS2 종료

File: documents/강사코드/ros2_ws_src/hello_cmake_pkg/src/talker.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class Talker : public rclcpp::Node {
public:
    Talker() : Node("talker"), count_(0) {
        publisher_ = this->create_publisher<std_msgs::msg::String>("hello_topic", 10);
        timer_ = this->create_wall_timer(
            std::chrono::milliseconds(500),
            std::bind(&Talker::timer_callback, this));
    }

private:
    void timer_callback() {
        auto msg = std_msgs::msg::String();
        msg.data = "Hello ROS2! [" + std::to_string(count_++) + "]";
        publisher_->publish(msg);
        RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", msg.data.c_str());
    }
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
    size_t count_;
};

int main(int argc, char * argv[]) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<Talker>());
    rclcpp::shutdown();
    return 0;
}
